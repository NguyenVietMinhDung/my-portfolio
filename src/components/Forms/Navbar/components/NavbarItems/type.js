import { Item } from '../../type';

export type NavbarItemsProps = {
  items: Array<Item>,
  activatedIndex: number,
  setActive: Function,
};
