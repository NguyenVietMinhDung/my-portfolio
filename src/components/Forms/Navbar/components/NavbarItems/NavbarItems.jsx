import React from 'react';
import PropTypes from 'prop-types';
import NavbarItem from '../NavbarItem';

const isActive = (index, activatedIndex) => index === activatedIndex;

const NavbarItems = (props) => {
  const { items, activatedIndex, setActive } = props;
  return (
    <div className="nav-items">
      {items.map((item, index) => (
        <NavbarItem
          index={index}
          active={isActive(index, activatedIndex)}
          key={item.name}
          name={item.name}
          url={item.url}
          setActive={setActive}
        />
      ))}
    </div>
  );
};

NavbarItems.propTypes = {
  items: PropTypes.instanceOf(Array),
  activatedIndex: PropTypes.number,
  setActive: PropTypes.func,
};

NavbarItems.defaultProps = {
  items: [],
  activatedIndex: 0,
  setActive: null,
};

export default NavbarItems;
