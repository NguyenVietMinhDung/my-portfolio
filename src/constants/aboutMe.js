export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
