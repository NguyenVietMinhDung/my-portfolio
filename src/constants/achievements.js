export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
