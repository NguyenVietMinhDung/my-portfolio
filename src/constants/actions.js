export const ACTIVATE_NAVIGATION_ITEM = 'ACTIVATE_NAVIGATION_ITEM';
export const OPEN_NAVIGATION_MENU = 'OPEN_NAVIGATION_MENU';
