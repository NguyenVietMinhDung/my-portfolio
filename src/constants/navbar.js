export const items = [
  {
    itemId: 'header',
    name: 'Home',
    url: '#home',
  },
  {
    itemId: 'about-me',
    name: 'About',
    url: '#about',
  },
  {
    itemId: 'work-experiences',
    name: 'Experiences',
    url: '#experiences',
  },
  {
    itemId: 'relevant-projects',
    name: 'Projects',
    url: '#projects',
  },
  {
    itemId: 'my-educations',
    name: 'Educations',
    url: '#educations',
  },
  {
    itemId: 'my-achievements',
    name: 'Achievements',
    url: '#achievements',
  },
  {
    itemId: 'footer',
    name: 'Contact',
    url: '#contact',
  },
];
export hamburgerIconUrl from '../assets/images/hamburger-icon.png';
export closeIconUrl from '../assets/images/close-icon.png';
