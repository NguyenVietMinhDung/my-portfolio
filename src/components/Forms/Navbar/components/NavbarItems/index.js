import NavbarItems from './NavbarItems';

export default NavbarItems;
