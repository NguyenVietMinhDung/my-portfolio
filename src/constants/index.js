import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
