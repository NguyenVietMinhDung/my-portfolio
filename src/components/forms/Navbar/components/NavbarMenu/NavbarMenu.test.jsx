import React from 'react';
import { shallow } from 'enzyme';
import { items } from '../../../../../constants/navbar';
import NavbarItems from './NavbarMenu';

const defaultProps = {
  items,
  activatedIndex: 0,
  isVisible: false,
  activateNavigationItem: jest.fn(),
};

describe('<NavbarItems />', () => {
  test('renders correctly', () => {
    const component = shallow(<NavbarItems {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });

  test('show navigation menu when clicking hamburger button', () => {
    const component = shallow(<NavbarItems {...defaultProps} isVisible />);
    expect(component.find('.is-visible').length).toEqual(1);
  });
});
