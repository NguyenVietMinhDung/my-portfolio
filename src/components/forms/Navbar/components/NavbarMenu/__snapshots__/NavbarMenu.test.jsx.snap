// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarItems /> renders correctly 1`] = `
<div
  className="nav-menu"
>
  <ul
    className="content is-invisible"
  >
    <Memo(NavbarItem)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      index={0}
      itemId="header"
      key="Home"
      name="Home"
      url="#home"
    />
    <Memo(NavbarItem)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      index={1}
      itemId="about-me"
      key="About"
      name="About"
      url="#about"
    />
    <Memo(NavbarItem)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      index={2}
      itemId="work-experiences"
      key="Experiences"
      name="Experiences"
      url="#experiences"
    />
    <Memo(NavbarItem)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      index={3}
      itemId="relevant-projects"
      key="Projects"
      name="Projects"
      url="#projects"
    />
    <Memo(NavbarItem)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      index={4}
      itemId="my-educations"
      key="Educations"
      name="Educations"
      url="#educations"
    />
    <Memo(NavbarItem)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      index={5}
      itemId="my-achievements"
      key="Achievements"
      name="Achievements"
      url="#achievements"
    />
    <Memo(NavbarItem)
      activateNavigationItem={[MockFunction]}
      activatedIndex={0}
      index={6}
      itemId="footer"
      key="Contact"
      name="Contact"
      url="#contact"
    />
  </ul>
</div>
`;
